datafusion/physical-expr/src/aggregate/array_agg_distinct.rs

```diff
@@ -151,7 +153,11 @@ impl Accumulator for DistinctArrayAggAccumulator {
             return Ok(());
         }
 
-        self.update_batch(states)
+        let array = &states[0];
+
+        // Unwrap outer ListArray then do update batch
+        let inner_array = array.as_list::<i32>().value(0);
```

Comment:
How do we know there is only a single row in the states array? Can we please add an assert here to make sure we don't accidentally lose data?

Something like `assert_eq!(array.len(), 1)`?